src/ensembl/src/content/app/browser/zmenu/ZmenuInstantDownload.tsx

```diff
@@ -15,74 +15,82 @@
  */
 
 import React from 'react';
+import { gql, useQuery } from '@apollo/client';
+import { Pick2 } from 'ts-multipick';
 
-import useApiService from 'src/shared/hooks/useApiService';
+import { parseFeatureId } from '../browserHelper';
 
 import { InstantDownloadTranscript } from 'src/shared/components/instant-download';
 import { CircleLoader } from 'src/shared/components/loader/Loader';
-import { TranscriptInResponse } from 'src/content/app/entity-viewer/shared/rest/rest-data-fetchers/transcriptData';
 
-import { LoadingState } from 'src/shared/types/loading-state';
+import { FullTranscript } from 'ensemblRoot/src/shared/types/thoas/transcript';
 
 import styles from './Zmenu.scss';
 
 type Props = {
   id: string;
 };
 
+const TRANSCRIPT_QUERY = gql`
+  query Transcript($genomeId: String!, $transcriptId: String!) {
+    transcript(byId: { genome_id: $genomeId, stable_id: $transcriptId }) {
+      metadata {
+        biotype {
+          value
+        }
+      }
+      gene {
+        stable_id
+      }
+    }
+  }
+`;
+
+type Transcript = Pick2<FullTranscript, 'metadata', 'biotype'> &
+  Pick2<FullTranscript, 'gene', 'stable_id'>;
+
 const ZmenuInstantDownload = (props: Props) => {
-  const genomeId = getGenomeId(props.id);
-  const transcriptId = getStableId(props.id);
-  const params = {
-    endpoint: `/lookup/id/${transcriptId}?content-type=application/json;expand=1`,
-    host: 'https://rest.ensembl.org'
-  };
-  const { loadingState, data, error } =
-    useApiService<TranscriptInResponse>(params);
+  const { genomeId, objectId: transcriptId } = parseFeatureId(props.id);
+
+  const { data, loading } = useQuery<{
+    transcript: Transcript;
+  }>(TRANSCRIPT_QUERY, {
+    variables: {
+      genomeId,
+      transcriptId
+    },
+    skip: !transcriptId
```

Comment:
Why are you treating `transcriptId` as if it can be undefined? The return type of `parseEnsObjectId` promises that the `objectId` field will always be present.